src/App.js

```diff
@@ -8,15 +8,21 @@ import "./App.css";
 function App() {
   const [{ user }, dispatch] = useDataLayerValue();
   const [loading, setLoading] = useState(true);
-  console.log({ user });
 
   useEffect(() => {
     auth.onAuthStateChanged((authUser) => {
       if (authUser) {
         if (!user) {
           dispatch({
             type: "SET_USER",
-            user: authUser.providerData[0],
+            user: {
+              uid: authUser.uid,
```

Comment:
you should deconstruct uid, displayName, email etc off authUser so you dont have object lookups every time.

const { uid, displayName } = authUser;

Reply:
done